src/main/scala/tech/cryptonomic/conseil/metadata/MetadataService.scala

```diff
@@ -1,48 +1,75 @@
 package tech.cryptonomic.conseil.metadata
 
+import scala.concurrent.duration._
 import tech.cryptonomic.conseil.config.Platforms.PlatformsConfiguration
-import tech.cryptonomic.conseil.generic.chain.DataTypes
+import tech.cryptonomic.conseil.generic.chain.{DataTypes, PlatformDiscoveryOperations}
 import tech.cryptonomic.conseil.generic.chain.PlatformDiscoveryTypes._
-import tech.cryptonomic.conseil.tezos.TezosPlatformDiscoveryOperations
-import tech.cryptonomic.conseil.util.CollectionOps.{ExtendedFuture, ExtendedOptionalList}
 import tech.cryptonomic.conseil.util.ConfigUtil
-import tech.cryptonomic.conseil.util.OptionUtil.when
 
 import scala.concurrent.Future.successful
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.language.postfixOps
 
 // service class for metadata
 class MetadataService(
     config: PlatformsConfiguration,
     transformation: UnitTransformation,
     cacheConfiguration: AttributeValuesCacheConfiguration,
-    tezosPlatformDiscoveryOperations: TezosPlatformDiscoveryOperations
-) {
+    platformDiscoveryOperations: PlatformDiscoveryOperations
+)(implicit apiExecutionContext: ExecutionContext) {
 
-  // checks if attribute is valid
-  def isAttributeValid(entity: String, attribute: String): Future[Boolean] =
-    tezosPlatformDiscoveryOperations.isAttributeValid(entity, attribute)
+  private val platforms = transformation.overridePlatforms(ConfigUtil.getPlatforms(config))
+
+  private val networks = platforms.map { platform =>
+    platform.path -> transformation.overrideNetworks(platform.path, ConfigUtil.getNetworks(config, platform.name))
+  }.toMap
+
+  private val entities = {
+    val allEntities = Await.result(platformDiscoveryOperations.getEntities, 1 second)
```

Comment:
That timeout is for a single call to the database. `1 sec` seems to be pretty long. I'm afraid of setting that value too high because when the database needs over 1 sec to answer, it usually means that something horribly happens and prefilling the cache won't succeed anyway so I try to fail fast. Anyway, we don't have metrics which could prove or disprove any point so I set it to 2 s. and we'll see what'll happen :)

Reply:
I was actually more interested in a solution that would handle the error and pass it back to the user with a clear information on what went wrong, but that might be better handled later?

Personally I think that the database might be unavailable for more than 1 sec with no critical situation going on, any network hiccup could bring unexpected consequences, or maybe a slightly overloaded system busy starting multiple services?
To be honest I didn't look at time tables to figure out what might be reasonable, and the only good answer would be to observe the system and measure any potential incident to extract real values.
tl;dr; I would've chosen something less strict but it's not really better than what you did, so let's say it's fine.